nucypher/characters/lawful.py

```diff
@@ -1255,15 +1276,27 @@ def run(self,
         elif start_reactor:  # ... without hendrix
             reactor.run()  # <--- Blocking Call (Reactor)
 
+    def is_running(self) -> bool:
+        """Returns True if this character has any running services currently."""
+        # TODO: Glean the status of all of Ursula's services.
+        # Sync up with start/stop logic ... it's a "service"
+        super_running = super().is_running()
+        return super_running
+
     def stop(self, halt_reactor: bool = False) -> None:
-        """Stop services"""
+        """
+        Stop services for partially or fully initialized characters.
+        # CAUTION #
+        """
         self.log.debug(f"---------Stopping {self}")
-        self._availability_tracker.stop()
-        self.stop_learning_loop()
-        if not self.federated_only:
-            self.work_tracker.stop()
-        if self._arrangement_pruning_task.running:
-            self._arrangement_pruning_task.stop()
+        # Handles the shutdown of a partially initialized character.
+        with contextlib.suppress(AttributeError):  # TODO: Is this acceptable here, what are alternatives?
+            self._availability_tracker.stop()
+            self.stop_learning_loop()
```

Comment:
Should `self.stop_learning_loop()` instead be `super().disenchant()` which runs `disenchant()` from `Character`? Ultimately both do the same thing of stopping the learning loop by calling `Learner.stop_learning_loop()` but fulfils the subclass usage...

Reply:
Hmm - yes this is worth consideration -- using some common shutdown logic methods.